### PR TITLE
chore: Provide a method on the Future for a flexible retry strategy 

### PR DESCRIPTION
**Pull request by jkwatson-verta**
## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [x] Other (explain) - tried this out in the registry for async http retries. 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

File: backend/common/src/main/java/ai/verta/modeldb/common/futures/DoublingBackoffStrategy.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@Getter(AccessLevel.NONE)
+class DoublingBackoffStrategy<R> implements RetryStrategy<R> {
+  private static final int INITIAL_DELAY_MILLIS = 10;
+  BiPredicate<R, Throwable> resultChecker;
+  int maxRetries;
+  AtomicInteger numberRetried = new AtomicInteger();
+
+  @Override
+  public Retry shouldRetry(R result, Throwable throwable) {
+    boolean apply = resultChecker.test(result, throwable);
```
Comment (eheinlein-verta) [inline backend/common/src/main/java/ai/verta/modeldb/common/futures/DoublingBackoffStrategy.java]: I'm not in love with the name `apply`. Maybe something like `success`?

File: backend/common/src/test/java/ai/verta/modeldb/common/futures/FutureTest.java [hunk 2]
```diff
@@ -144,6 +145,57 @@ void retry() throws Exception {
...
 
+  @Test
+  void retry_strategy_failure() {
+    AtomicInteger sequencer = new AtomicInteger();
+
+    assertThatThrownBy(
+            () ->
+                Future.retrying(
+                        () -> {
+                          if (sequencer.getAndIncrement() < 10) {
```
Comment (eheinlein-verta) [inline backend/common/src/test/java/ai/verta/modeldb/common/futures/FutureTest.java]: Is the goal of the # 10 here to just be more than 6 (the expected max number)? Or is it more specific? If the goal here is just to be "too many retries" maybe pull the literal into a variable named `tooManyRetries` or something?

**Review by eheinlein-verta — APPROVED** [review]
Approved, with just some small naming suggestions if you agree.

File: backend/common/src/main/java/ai/verta/modeldb/common/futures/DoublingBackoffStrategy.java (hunk 1 above)
Comment (jkwatson-verta) [reply backend/common/src/main/java/ai/verta/modeldb/common/futures/DoublingBackoffStrategy.java]: yes. this was left over from a refactoring. will definitely clean up. good catch!

File: backend/common/src/test/java/ai/verta/modeldb/common/futures/FutureTest.java (hunk 2 above)
Comment (jkwatson-verta) [reply backend/common/src/test/java/ai/verta/modeldb/common/futures/FutureTest.java]: yeah, it's just intended to be more times than the maximum configured. I'll rename it. good suggestion!

**Pull request merged**